africanus/gridding/simple/gridding.py

```diff
@@ -43,142 +54,96 @@ def _grid(vis, uvw, flags, ref_wave, nx, ny, convolution_filter):
+            disc_u_psf = int(np.round(twice_scaled_u))
+            disc_v_psf = int(np.round(twice_scaled_v))
 
+            base_frac_u_psf = one_half_sup + disc_u_psf - twice_scaled_u
+            base_frac_v_psf = one_half_sup + disc_v_psf - twice_scaled_v
 
-def _psf(vis, uvw, flags, ref_wave, nx, ny, convolution_filter):
-    """
-    Convolutional PSF gridder (continuum)
+            frac_u_psf = int(base_frac_u_psf*cf.oversample)
+            frac_v_psf = int(base_frac_v_psf*cf.oversample)
 
-    Parameters
-    ----------
-    vis : np.ndarray
-        complex64 visibility array of shape (row, chan, corr)
-    uvw : np.ndarray
-        float64 array of UVW coordinates of shape (row, 3)
-    flags : np.ndarray
-        flagged data array of shape (row, chan, corr).
-        Any positive quantity will indicate that the corresponding
-        visibility should be flagged
-    ref_wave : np.ndarray
-        float64 array of wavelengths of shape (chan,)
-    nx : integer
-        Size of the grid's X dimension
-    ny : integer
-        Size of the grid's Y dimension
-    convolution_filter :  :class:`~africanus.filters.ConvolutionFilter`
-        Anti-Aliasing filter object
-
-    Returns
-    -------
-    np.ndarray
-        (nx, ny) containing the PSF, constructed
-        by summing convolutional weights.
-    """
-    cf = convolution_filter
-
-    assert vis.shape[1] == ref_wave.shape[0]
-    filter_index = np.arange(-cf.half_sup, cf.half_sup+1)
-
-    # for deconvolution the PSF should be 2x size of the image (see
-    # Hogbom CLEAN for details), one grid for the sampling function:
-    grid = np.zeros((2*ny,2*nx), dtype=np.complex64)
-    rgrid = grid.real
-
-    for r in range(uvw.shape[0]):                 # row (vis)
-        for c in range(vis.shape[1]):             # channel
-            # Continue if any correlation is flagged
-            flagged = False
-
-            for p in range(vis.shape[2]):
-                if flags[r,c,p] > 0:
-                    flagged = True
-                    break
+            # Iterate over v/y
+            for conv_v in filter_index:
+                base_v = conv_v*cf.oversample
+                v_tap = cf.filter_taps[base_v + frac_v]
+                v_tap_psf = cf.filter_taps[base_v + frac_v_psf]
 
-            if flagged:
-                continue
+                grid_v = disc_v + conv_v + ny // 2
+                grid_v_psf = disc_v_psf + conv_v + ny
 
-            scaled_v = uvw[r,0] / ref_wave[c]
-            scaled_u = uvw[r,1] / ref_wave[c]
+                # Iterate over u/x
+                for conv_u in filter_index:
+                    base_u = conv_u*cf.oversample
+                    u_tap = cf.filter_taps[base_u + frac_u]
+                    u_tap_psf = cf.filter_taps[base_u + frac_u_psf]
 
-            disc_u = int(round(scaled_u))
-            disc_v = int(round(scaled_v))
+                    conv_weight = v_tap * u_tap
```

Comment:
Looks like we are assuming a WEIGHT column of all ones. We should incorporate a weighting scheme. I think just natural weighting will be fine, at least for what I am planning to use the gridder for. But that is a separate pull request...